PatternPal/PatternPal/Services/StepByStepService.cs

```diff
@@ -1,41 +1,211 @@
+        else
+        {
+            selectedNodePath = request.Documents[0];
+        }
+        
+        RecognizerRunner runner = new(
+            selectedNodePath,
+            InstructionSetHolder.instructionSet.Steps[request.InstructionId]);
+        IList<ICheckResult> res = runner.Run(pruneAll: true);
+        NodeCheckResult checkRes = (NodeCheckResult)res[0];
+
+        return Task.FromResult(new CheckInstructionResponse{ Result = checkRes.ChildrenCheckResults.Count != 0});
+    }
+
+    // TODO Method may still be useful depending on the presentation in the detector view
+    /// <inheritdoc />
+    private static CheckResult CreateCheckResult(
+        ICheckResult checkResult)
     {
         throw new NotImplementedException();
+        //CheckResult newCheckResult = new()
+        //{
+        //    FeedbackType = (CheckResult.Types.FeedbackType)((int)checkResult.GetFeedbackType() + 1),
+        //    Hidden = checkResult.IsHidden,
+        //    FeedbackMessage = ResourceUtils.ResultToString(checkResult),
+        //};
+        //foreach (Recognizers.Abstractions.ICheckResult childCheckResult in checkResult.GetChildFeedback())
+        //{
+        //    newCheckResult.SubCheckResults.Add(CreateCheckResult(childCheckResult));
+        //}
+        //return newCheckResult;
     }
 
-    private static class State
+    /// <inheritdoc />
+    public override Task< SetFilePathResponse > SetNewFilePath(
+        SetFilePathRequest request,
+        ServerCallContext context)
     {
-        internal static readonly Dictionary< string, string > StateKeyed = new();
+        //throw new NotImplementedException();
+        string filepath = request.FilePath;
+        foreach (IInstruction current in InstructionSetHolder.instructionSet.Steps)
+        {
+            current.FileId = filepath;
+        }
+
+        var test = InstructionSetHolder.instructionSet;
+
+        return Task.FromResult(new SetFilePathResponse());
     }
+    
+    public IRecognizer GetRecognizer(Recognizer recognizer)
+    {
+        IDictionary< Recognizer, IRecognizer > supportedRecognizers =
+            new Dictionary< Recognizer, IRecognizer >();
+
+        Type recognizerType = typeof( IRecognizer );
+
+        // Find all types which derive from `IRecognizer`.
+        foreach (Type type in recognizerType.Assembly.GetTypes().Where(ty => ty != recognizerType && recognizerType.IsAssignableFrom(ty)))
+        {
+            IRecognizer instance = (IRecognizer)Activator.CreateInstance(type)!;
+
+            // Get the mapping from `Recognizer` to `IRecognizer` by invoking the property on the
+            // recognizer which specifies it.
+            supportedRecognizers.Add(
+                (Recognizer)type.GetRuntimeProperty(nameof( IRecognizer.RecognizerType ))!.GetValue(instance)!,
+                instance);
+        }
+        return supportedRecognizers[ recognizer ];
+    }
+}
+
+public static class InstructionSetHolder
```

Comment:
What is the point of this class? When you receive a `CheckInstruction` request, can't you just lookup the `InstructionSet` to use based on the provided info?

Reply:
See earlier comment about stateless.